{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dependencies\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ask user to enter directory path for gene file \n",
    "gene_file_path_input = input(\"enter path of working directory: \") # ask user ot enter gene file path\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Smell\\tSmell-1\\tchr1\\t100\\t1000\\t+\\t(100-150,400-500,900-1000)\\n',\n",
       " 'Smell\\tSmell-2\\tchr1\\t100\\t1000\\t+\\t(100-150,900-1000)\\n',\n",
       " 'Size\\tSize-1\\tchr1\\t1500\\t2000\\t+\\t(1500-1750,1800-1850,\\t1900-2000)\\n',\n",
       " 'Size\\tSize-2\\tchr1\\t1500\\t2000\\t+\\t(1500-2000)\\n',\n",
       " 'Color\\tColor-1\\tchr2\\t3400\\t4200\\t-\\t(3400-3600,4000-4200)\\n',\n",
       " 'Color\\tColor-2\\tchr2\\t3400\\t4200\\t-\\t(3400-3550,3800-3900,4000-4200)\\n',\n",
       " 'Color\\tColor-3\\tchr2\\t3400\\t4200\\t-\\t(3400-3600,3800-3900,4100-4200)\\n',\n",
       " 'appl4\\tappl4-1\\tchr3\\t50\\t800\\t+\\t(50-400,700-80)\\n',\n",
       " 'appl5\\tappl5-1\\tchr3\\t1100\\t1500\\t+\\t(1100-1200,1450-1500)\\n',\n",
       " 'Taste\\tTaste-1\\tchr3\\t2000\\t2600\\t-\\t(2000-2300,2500-2600)\\n',\n",
       " 'Taste\\tTaste-2\\tchr3\\t2000\\t2600\\t-\\t(2000-2050,2200-2300,2500-2600)\\n',\n",
       " 'Shape\\tShape-1\\tchr4\\t3100\\t3700\\t-\\t(3100-3500,3600-3700)\\n',\n",
       " 'Shape\\tShape-2\\tchr4\\t3100\\t3700\\t-\\t(3100-3200,3400-3500,3600-3700)\\n',\n",
       " 'appl8\\tappl8-1\\tchr4\\t4500\\t4800\\t+\\t(4500-4800)\\n',\n",
       " 'appl9\\tappl9-1\\tchr5\\t200\\t1000\\t-\\t(200-450,550-650,800-1000)\\n',\n",
       " 'appl10\\tappl10-1\\tchr5\\t1100\\t1700\\t+\\t(1100-1400,1500-1700)\\n']"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# open and read gene file\n",
    "gene_file = open(gene_file_path_input, \"r\") \n",
    "gene_file = gene_file.readlines()\n",
    "gene_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>chr_num</th>\n",
       "      <th>gene_start</th>\n",
       "      <th>gene_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Smell</td>\n",
       "      <td>Smell-1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>(100-150,400-500,900-1000)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Smell</td>\n",
       "      <td>Smell-2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100</td>\n",
       "      <td>1000</td>\n",
       "      <td>+</td>\n",
       "      <td>(100-150,900-1000)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Size</td>\n",
       "      <td>Size-1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1500</td>\n",
       "      <td>2000</td>\n",
       "      <td>+</td>\n",
       "      <td>(1500-1750,1800-1850,1900-2000)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Size</td>\n",
       "      <td>Size-2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1500</td>\n",
       "      <td>2000</td>\n",
       "      <td>+</td>\n",
       "      <td>(1500-2000)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Color</td>\n",
       "      <td>Color-1</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3400</td>\n",
       "      <td>4200</td>\n",
       "      <td>-</td>\n",
       "      <td>(3400-3600,4000-4200)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Color</td>\n",
       "      <td>Color-2</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3400</td>\n",
       "      <td>4200</td>\n",
       "      <td>-</td>\n",
       "      <td>(3400-3550,3800-3900,4000-4200)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Color</td>\n",
       "      <td>Color-3</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3400</td>\n",
       "      <td>4200</td>\n",
       "      <td>-</td>\n",
       "      <td>(3400-3600,3800-3900,4100-4200)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>appl4</td>\n",
       "      <td>appl4-1</td>\n",
       "      <td>chr3</td>\n",
       "      <td>50</td>\n",
       "      <td>800</td>\n",
       "      <td>+</td>\n",
       "      <td>(50-400,700-80)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>appl5</td>\n",
       "      <td>appl5-1</td>\n",
       "      <td>chr3</td>\n",
       "      <td>1100</td>\n",
       "      <td>1500</td>\n",
       "      <td>+</td>\n",
       "      <td>(1100-1200,1450-1500)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Taste</td>\n",
       "      <td>Taste-1</td>\n",
       "      <td>chr3</td>\n",
       "      <td>2000</td>\n",
       "      <td>2600</td>\n",
       "      <td>-</td>\n",
       "      <td>(2000-2300,2500-2600)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Taste</td>\n",
       "      <td>Taste-2</td>\n",
       "      <td>chr3</td>\n",
       "      <td>2000</td>\n",
       "      <td>2600</td>\n",
       "      <td>-</td>\n",
       "      <td>(2000-2050,2200-2300,2500-2600)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Shape</td>\n",
       "      <td>Shape-1</td>\n",
       "      <td>chr4</td>\n",
       "      <td>3100</td>\n",
       "      <td>3700</td>\n",
       "      <td>-</td>\n",
       "      <td>(3100-3500,3600-3700)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Shape</td>\n",
       "      <td>Shape-2</td>\n",
       "      <td>chr4</td>\n",
       "      <td>3100</td>\n",
       "      <td>3700</td>\n",
       "      <td>-</td>\n",
       "      <td>(3100-3200,3400-3500,3600-3700)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>appl8</td>\n",
       "      <td>appl8-1</td>\n",
       "      <td>chr4</td>\n",
       "      <td>4500</td>\n",
       "      <td>4800</td>\n",
       "      <td>+</td>\n",
       "      <td>(4500-4800)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>appl9</td>\n",
       "      <td>appl9-1</td>\n",
       "      <td>chr5</td>\n",
       "      <td>200</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>(200-450,550-650,800-1000)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>appl10</td>\n",
       "      <td>appl10-1</td>\n",
       "      <td>chr5</td>\n",
       "      <td>1100</td>\n",
       "      <td>1700</td>\n",
       "      <td>+</td>\n",
       "      <td>(1100-1400,1500-1700)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene   variant chr_num gene_start gene_end strand  \\\n",
       "0    Smell   Smell-1    chr1        100     1000      +   \n",
       "1    Smell   Smell-2    chr1        100     1000      +   \n",
       "2     Size    Size-1    chr1       1500     2000      +   \n",
       "3     Size    Size-2    chr1       1500     2000      +   \n",
       "4    Color   Color-1    chr2       3400     4200      -   \n",
       "5    Color   Color-2    chr2       3400     4200      -   \n",
       "6    Color   Color-3    chr2       3400     4200      -   \n",
       "7    appl4   appl4-1    chr3         50      800      +   \n",
       "8    appl5   appl5-1    chr3       1100     1500      +   \n",
       "9    Taste   Taste-1    chr3       2000     2600      -   \n",
       "10   Taste   Taste-2    chr3       2000     2600      -   \n",
       "11   Shape   Shape-1    chr4       3100     3700      -   \n",
       "12   Shape   Shape-2    chr4       3100     3700      -   \n",
       "13   appl8   appl8-1    chr4       4500     4800      +   \n",
       "14   appl9   appl9-1    chr5        200     1000      -   \n",
       "15  appl10  appl10-1    chr5       1100     1700      +   \n",
       "\n",
       "                      gene_location  \n",
       "0        (100-150,400-500,900-1000)  \n",
       "1                (100-150,900-1000)  \n",
       "2   (1500-1750,1800-1850,1900-2000)  \n",
       "3                       (1500-2000)  \n",
       "4             (3400-3600,4000-4200)  \n",
       "5   (3400-3550,3800-3900,4000-4200)  \n",
       "6   (3400-3600,3800-3900,4100-4200)  \n",
       "7                   (50-400,700-80)  \n",
       "8             (1100-1200,1450-1500)  \n",
       "9             (2000-2300,2500-2600)  \n",
       "10  (2000-2050,2200-2300,2500-2600)  \n",
       "11            (3100-3500,3600-3700)  \n",
       "12  (3100-3200,3400-3500,3600-3700)  \n",
       "13                      (4500-4800)  \n",
       "14       (200-450,550-650,800-1000)  \n",
       "15            (1100-1400,1500-1700)  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# clean gene file and convert to panda DataFrame\n",
    "gene_clean_data = [i.replace (\",\\t\", \",\").strip('\\n').split('\\t') for i in gene_file]\n",
    "col_names = ['gene', 'variant', 'chr_num', 'gene_start', 'gene_end','strand', 'gene_location']\n",
    "df = pd.DataFrame(gene_clean_data, columns = col_names)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Smell', 'Size', 'Color', 'appl4', 'appl5', 'Taste', 'Shape',\n",
       "       'appl8', 'appl9', 'appl10'], dtype=object)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find uniqe genes\n",
    "unique_genes = df['gene'].unique()\n",
    "unique_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Color</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shape</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Size</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Smell</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Taste</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl10</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl5</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl8</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl9</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        variant\n",
       "gene           \n",
       "Color         3\n",
       "Shape         2\n",
       "Size          2\n",
       "Smell         2\n",
       "Taste         2\n",
       "appl10        1\n",
       "appl4         1\n",
       "appl5         1\n",
       "appl8         1\n",
       "appl9         1"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find variant per gene\n",
    "gene_variant_count = pd.DataFrame(df.groupby(\"gene\")[\"variant\"].count())\n",
    "gene_variant_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>appl10</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl5</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl8</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>appl9</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        variant\n",
       "gene           \n",
       "appl10        1\n",
       "appl4         1\n",
       "appl5         1\n",
       "appl8         1\n",
       "appl9         1"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find genes with single variant\n",
    "single_variant_genes= gene_variant_count[(gene_variant_count == 1).any(axis=1)]\n",
    "single_variant_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Color</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shape</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Size</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Smell</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Taste</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       variant\n",
       "gene          \n",
       "Color        3\n",
       "Shape        2\n",
       "Size         2\n",
       "Smell        2\n",
       "Taste        2"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find genes with multiple variants\n",
    "multiple_variants_genes= gene_variant_count[(gene_variant_count != 1).any(axis=1)]\n",
    "multiple_variants_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strand</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>+</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>-</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene\n",
       "strand      \n",
       "+          8\n",
       "-          8"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# genes per ‘+’ and ‘-’ strand\n",
    "strand_gene_count = pd.DataFrame(df.groupby(\"strand\")[\"gene\"].count())\n",
    "strand_gene_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_num</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene\n",
       "chr_num      \n",
       "chr1        4\n",
       "chr2        3\n",
       "chr3        4\n",
       "chr4        3\n",
       "chr5        2"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find genes per chromosome\n",
    "chromosome_gene_count = pd.DataFrame(df.groupby(\"chr_num\")[\"gene\"].count())\n",
    "chromosome_gene_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_num</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         variant\n",
       "chr_num         \n",
       "chr1           4\n",
       "chr2           3\n",
       "chr3           4\n",
       "chr4           3\n",
       "chr5           2"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find variant per chromosome\n",
    "chromosome_variant_count = pd.DataFrame(df.groupby(\"chr_num\")[\"variant\"].count())\n",
    "chromosome_variant_count"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"
  },
  "kernelspec": {
   "display_name": "Python 3.9.12 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
